jujugui/static/gui/src/app/jujulib/index.js

```diff
@@ -57,9 +57,17 @@ var module = module;
       }
       callback(error, data);
     };
-    var failure = function(xhr) {
-      var data = JSON.parse(xhr.target.responseText);
-      var error = data.Message || data.message || data.Error || data.error;
+    var failure = function(xhrOrMessage) {
```

Comment:
It would probably be nice to have a proper docblock for this fn with a description of the values that could be passed in.